modules/postgresql/variables.tf

```diff
@@ -128,143 +148,175 @@ variable "read_replica_size" {
+  type        = string
   default     = ""
 }
 
 variable "read_replica_tier" {
   description = "The tier for the read replica instances."
+  type        = string
   default     = ""
 }
 
 variable "read_replica_zones" {
   description = "The zones for the read replica instancess, it should be something like: `a,b,c`. Given zones are used rotationally for creating read replicas."
+  type        = string
   default     = ""
 }
 
 variable "read_replica_activation_policy" {
   description = "The activation policy for the read replica instances.Can be either `ALWAYS`, `NEVER` or `ON_DEMAND`."
+  type        = string
   default     = "ALWAYS"
 }
 
 variable "read_replica_availability_type" {
   description = "The availability type for the read replica instances.This is only used to set up high availability for the PostgreSQL instances. Can be either `ZONAL` or `REGIONAL`."
+  type        = string
   default     = "ZONAL"
 }
 
 variable "read_replica_crash_safe_replication" {
   description = "The crash safe replication is to indicates when crash-safe replication flags are enabled."
+  type        = bool
   default     = true
 }
 
 variable "read_replica_disk_autoresize" {
   description = "Configuration to increase storage size."
+  type        = bool
   default     = true
 }
 
 variable "read_replica_disk_size" {
   description = "The disk size for the read replica instances."
+  type        = number
   default     = 10
 }
 
 variable "read_replica_disk_type" {
   description = "The disk type for the read replica instances."
+  type        = string
   default     = "PD_SSD"
 }
 
 variable "read_replica_pricing_plan" {
   description = "The pricing plan for the read replica instances."
+  type        = string
   default     = "PER_USE"
 }
 
 variable "read_replica_maintenance_window_day" {
   description = "The day of week (1-7) for the read replica instances maintenance."
+  type        = number
   default     = 1
 }
 
 variable "read_replica_maintenance_window_hour" {
   description = "The hour of day (0-23) maintenance window for the read replica instances maintenance."
+  type        = number
   default     = 23
 }
 
 variable "read_replica_maintenance_window_update_track" {
   description = "The update track of maintenance window for the read replica instances maintenance.Can be either `canary` or `stable`."
+  type        = string
   default     = "canary"
 }
 
 variable "read_replica_database_flags" {
   description = "The database flags for the read replica instances. See [more details](https://cloud.google.com/sql/docs/mysql/flags)"
+  type        = list(map(string))
   default     = []
 }
 
 variable "read_replica_configuration" {
-  default     = {}
-  description = "The replica configuration for use in read replica"
+  description = "The replica configuration for use in all read replica instances."
+  type = object({
+    connect_retry_interval = number
+    dump_file_path         = string
+  })
+  default = null
```

Comment:
using type = object () makes it required, and if send any param in a non-replication instance, it will break returning invalid request.
I'm not sure about that, but I think it should work with an empty map by default.

Reply:
I don’t think so, because of different types of variables used in it